x/ecocredit/server/core/msg_server.go

```diff
@@ -0,0 +1,1200 @@
+			Canceller:  holder.String(),
+			BatchDenom: credit.BatchDenom,
+			Amount:     credit.Amount,
+		}); err != nil {
+			return nil, err
+		}
+		sdkCtx.GasMeter().ConsumeGas(gasCostPerIteration, "cancel ecocredits")
+	}
+	return &v1beta1.MsgCancelResponse{}, nil
+}
+
+func (s serverImpl) UpdateClassAdmin(ctx context.Context, req *v1beta1.MsgUpdateClassAdmin) (*v1beta1.MsgUpdateClassAdminResponse, error) {
+	classInfo, err := s.stateStore.ClassInfoStore().GetByName(ctx, req.ClassId)
+	if err != nil {
+		return nil, err
+	}
+	reqAddr, _ := sdk.AccAddressFromBech32(req.Admin)
+	classAdmin := sdk.AccAddress(classInfo.Admin)
+	if !classAdmin.Equals(reqAddr) {
+		return nil, sdkerrors.ErrUnauthorized.Wrapf("expected admin %s, got %s", classInfo.Admin, req.Admin)
+	}
+	classInfo.Admin = reqAddr
+	if err = s.stateStore.ClassInfoStore().Update(ctx, classInfo); err != nil {
+		return nil, err
+	}
+	return &v1beta1.MsgUpdateClassAdminResponse{}, err
+}
+
+func (s serverImpl) UpdateClassIssuers(ctx context.Context, req *v1beta1.MsgUpdateClassIssuers) (*v1beta1.MsgUpdateClassIssuersResponse, error) {
+	class, err := s.stateStore.ClassInfoStore().GetByName(ctx, req.ClassId)
+	if err != nil {
+		return nil, err
+	}
+	reqAddr, _ := sdk.AccAddressFromBech32(req.Admin)
+	admin := sdk.AccAddress(class.Admin)
+	if !reqAddr.Equals(admin) {
+		return nil, sdkerrors.ErrUnauthorized.Wrapf("expected admin %s, got %s", class.Admin, req.Admin)
+	}
+
+	// delete the old issuers
+	if err = s.stateStore.ClassIssuerStore().DeleteBy(ctx, ecocreditv1beta1.ClassIssuerClassIdIssuerIndexKey{}.WithClassId(class.Name)); err != nil {
+		return nil, err
+	}
+
+	// add the new issuers
+	for _, issuer := range req.Issuers {
+		iAddr, _ := sdk.AccAddressFromBech32(issuer)
+		if err = s.stateStore.ClassIssuerStore().Insert(ctx, &ecocreditv1beta1.ClassIssuer{
+			ClassId: req.ClassId,
+			Issuer:  iAddr,
+		}); err != nil {
+			return nil, err
+		}
+	}
+	return &v1beta1.MsgUpdateClassIssuersResponse{}, nil
+}
+
+func (s serverImpl) UpdateClassMetadata(ctx context.Context, req *v1beta1.MsgUpdateClassMetadata) (*v1beta1.MsgUpdateClassMetadataResponse, error) {
+	classInfo, err := s.stateStore.ClassInfoStore().GetByName(ctx, req.ClassId)
+	if err != nil {
+		return nil, err
+	}
+	reqAddr, _ := sdk.AccAddressFromBech32(req.Admin)
+	admin := sdk.AccAddress(classInfo.Admin)
+	if !reqAddr.Equals(admin) {
+		return nil, sdkerrors.ErrUnauthorized.Wrapf("expected admin %s, got %s", classInfo.Admin, req.Admin)
+	}
+	classInfo.Metadata = req.Metadata
+	if err = s.stateStore.ClassInfoStore().Update(ctx, classInfo); err != nil {
+		return nil, err
+	}
+	return &v1beta1.MsgUpdateClassMetadataResponse{}, err
+}
+
+// Sell creates new sell orders for credits
+// TODO: impl with ORM https://github.com/regen-network/regen-ledger/issues/728
+func (s serverImpl) Sell(ctx context.Context, sell *marketplacev1beta1.MsgSell) (*marketplacev1beta1.MsgSellResponse, error) {
```

Comment:
i think we can just delete all this marketplace stuff and not have this implement marketplacev1beta1.MsgServer at all

Reply:
deleted